helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ResourceMonitor.java

```diff
@@ -370,10 +388,10 @@ public void updateStateHandoffStats(MonitorState monitorState, long totalDuratio
   public void updateRebalancerStats(long numPendingRecoveryRebalancePartitions,
       long numPendingLoadRebalancePartitions, long numRecoveryRebalanceThrottledPartitions,
       long numLoadRebalanceThrottledPartitions) {
-    _numPendingRecoveryRebalancePartitions.updateValue(numPendingRecoveryRebalancePartitions);
-    _numPendingLoadRebalancePartitions.updateValue(numPendingLoadRebalancePartitions);
-    _numRecoveryRebalanceThrottledPartitions.updateValue(numRecoveryRebalanceThrottledPartitions);
-    _numLoadRebalanceThrottledPartitions.updateValue(numLoadRebalanceThrottledPartitions);
+    _numPendingRecoveryRebalanceReplicas.updateValue(numPendingRecoveryRebalancePartitions);
```

Comment:
Why leave all the deprecated fields? They will be zero forever, right?
I prefer removing them. Or if strict backward compatibility is to be ensured, then just keep updating their values.